client/rpc_test.go

```diff
@@ -191,3 +192,57 @@ func Test_resolveServer(t *testing.T) {
 	}
 
 }
+
+func TestRpc_RetryBlockTime(t *testing.T) {
```

Comment:
A test!!! :tada:

Reply:
The test took 10x longer to write than the fix, unfortunately. Not having any way of controlling the behavior of the lower layers of the RPC "stack" we have is probably why we have fairly poor test coverage of the error handling paths. :crying_cat_face: